internal/fs/fs.go

```diff
@@ -2138,7 +2138,6 @@ func (fs *fileSystem) OpenDir(
 	ctx context.Context,
 	op *fuseops.OpenDirOp) (err error) {
 	fs.mu.Lock()
-	defer fs.mu.Unlock()
```

Comment:
what happens if line no: 2146 fails, will the fs.mu gets unlocked?

Reply:
Based on the documentation: https://github.com/golang/go/blob/87ec2c959c73e62bfae230ef7efca11ec2a90804/src/builtin/builtin.go#L265, the defer won't execute in this case but IIUC I think it should be okay because we are panicking in that case and as soon as the go-routine finishes, GCSFuse will terminate.

Reply:
+1
Any failure in line no: 2146 leads to panic, which we handle in https://github.com/GoogleCloudPlatform/gcsfuse/blob/cecf4ac1902791d21bafcb87fceae0002db3c57f/internal/fs/wrappers/error_mapping.go#L92 and terminate the gcsfuse process.